app/classifier/index.cjsx

```diff
@@ -51,8 +78,14 @@ Classifier = React.createClass
       @prepareToClassify nextProps.classification
 
     @context.geordi.remember subjectID: @props.subject?.id
+    @reCheckIfInterventionNeeded()
+
+  componentWillMount: () ->
+    @context.interventionMonitor.setProjectSlug @props.project.slug
 
   componentWillUnmount: () ->
+    @context.interventionMonitor.removeListener 'interventionRequested', => {}
```

Comment:
The second argument here should exactly match the listener function that you want to remove. This code leaves your original event listener function in place even after the component has unmounted.